datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/contract/DataContractTab.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useGetDatasetContractQuery } from '../../../../../../../graphql/contract.generated';
+import { DataContractState } from '../../../../../../../types.generated';
+import { useEntityData } from '../../../../EntityContext';
+import { DataContractEmptyState } from './DataContractEmptyState';
+import { DataContractSummary } from './DataContractSummary';
+import { DataQualityContractSummary } from './DataQualityContractSummary';
+import { SchemaContractSummary } from './SchemaContractSummary';
+import { FreshnessContractSummary } from './FreshnessContractSummary';
+import { DataContractBuilderModal } from './builder/DataContractBuilderModal';
+import { createBuilderState } from './builder/utils';
+import { getAssertionsSummary } from '../utils';
+
+const Container = styled.div`
+    display: flex;
+`;
+
+const LeftColumn = styled.div`
+    width: 50%;
+`;
+
+const RightColumn = styled.div`
+    width: 50%;
+`;
+
+/**
+ * Component used for rendering the Data Contract Tab on the Assertions parent tab.
+ */
+export const DataContractTab = () => {
+    const { urn, entityType } = useEntityData();
+
+    const { data, refetch } = useGetDatasetContractQuery({
+        variables: {
+            urn,
+        },
+    });
+    const [showContractBuilder, setShowContractBuilder] = useState(false);
+
+    const contract = data?.dataset?.contract;
+    const schemaContracts = data?.dataset?.contract?.properties?.schema || [];
+    const freshnessContracts = data?.dataset?.contract?.properties?.freshness || [];
+    const dataQualityContracts = data?.dataset?.contract?.properties?.dataQuality || [];
+    const schemaAssertions = schemaContracts.map((c) => c.assertion);
+    const freshnessAssertions = freshnessContracts.map((c) => c.assertion);
+    const dataQualityAssertions = dataQualityContracts.map((c) => c.assertion);
+    const assertionsSummary = getAssertionsSummary([
+        ...schemaAssertions,
+        ...freshnessAssertions,
+        ...dataQualityAssertions,
+    ] as any);
+    const contractState = data?.dataset?.contract?.status?.state || DataContractState.Active;
+    const hasFreshnessContract = freshnessContracts && freshnessContracts?.length;
+    const hasSchemaContract = schemaContracts && schemaContracts?.length;
+    const hasDataQualityContract = dataQualityContracts && dataQualityContracts?.length;
+    const showLeftColumn = hasFreshnessContract || hasSchemaContract || undefined;
+
+    const onContractUpdate = () => {
+        if (contract) {
+            // Contract exists, just refetch.
+            refetch();
+        } else {
+            // no contract yet, wait for indxing,
+            setTimeout(() => refetch(), 3000);
+        }
+        setShowContractBuilder(false);
+    };
+
+    return (
+        <>
+            {(data?.dataset?.contract && (
```

Comment:
should use a ternary check:
`{data?.dataset?.contract ? (...) : <DataContractEmptyState .../>}`

Reply:
Also would copy these changes to SaaS as we don't have an 'EmptyState' component there and I think we are also not using ternaries there

Reply:
Done for both